src/transformStateWithClones.js

```diff
@@ -6,8 +6,46 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  const stateCopy = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        // add all extraData props to the state
+        Object.assign(stateCopy, action.extraData);
+
+        break;
+
+      case 'removeProperties': {
+        // remove all keysToRemove from the state
+        for (const key of action.keysToRemove) {
+          delete stateCopy[key];
+        }
+
+        break;
+      }
+
+      case 'clear': {
+        // remove all state keys from it
+        for (const key in stateCopy) {
+          delete stateCopy[key];
+        }
+
+        break;
+      }
+
+      default: {
+        break;
+      }
```

Comment:
It's better to handle errors in default block, for example you can throw an error with message like "Unknown action type"

Reply:
Added some changes. 
Is this update okay?